src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.search
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.usages.PsiNamedElementUsageGroupBase
+import com.intellij.usages.Usage
+import com.intellij.usages.UsageGroup
+import com.intellij.usages.UsageTarget
+import com.intellij.usages.impl.FileStructureGroupRuleProvider
+import com.intellij.usages.rules.PsiElementUsage
+import com.intellij.usages.rules.SingleParentUsageGroupingRule
+import com.intellij.usages.rules.UsageGroupingRule
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+
+class RsFunctionGroupingRuleProvider : FileStructureGroupRuleProvider {
+    override fun getUsageGroupingRule(project: Project): UsageGroupingRule? {
+        return RsFunctionGroupingRule()
+    }
+
+    private class RsFunctionGroupingRule : SingleParentUsageGroupingRule() {
```

Comment:
Make it staitc (move to `companion object { ... }`).

Reply:
@mchernyavsky Are you about `RsFunctionGroupingRule` class?
If yes then it's already static